### PR TITLE
v5.2: Integrate mvn archetype module 'ksql-udf-quickstart' into project

### PR DESCRIPTION
**Pull request by big-andy-coates**
### Description 

This PR replaces #2613

update mvn archetype to better fit with project build lifecycle:
- include module in project
- standardised the module pom
- current project version is now injected into the archetype.
- improved generated example code
  - to meet our coding standards
  - to set author and version on generated UD(A)Fs.
  - to use @UdfParameter in UDFs.
- added integration test to ensure archetype project will build

### Testing done 

Added integration test to test projects built from the archetype actually compile.
Manual testing of project's built from archetype.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

**Comment by mitch-seymour** [comment]
This looks great, thanks for improving this :)

File: ksql-udf-quickstart/src/main/resources/archetype-resources/src/main/java/ReverseUdf.java [hunk 1]
```diff
@@ -17,27 +17,45 @@
...
+    description = "Example UDF that reverses an object",
+    version = "${version}",
+    author = "${author}"
+)
 public class ReverseUdf {
 
   @Udf(description = "Reverse a string")
-  public String reverseString(String source) {
+  public String reverseString(
+      @UdfParameter(value = "source", description = "the value to reverse")
```
Comment (spena) [inline ksql-udf-quickstart/src/main/resources/archetype-resources/src/main/java/ReverseUdf.java]: If the value is omitted, then the parameter name will use the same variable name. 
Should we take advantage of that? It is not on the UDF doc either.

**Comment by mitch-seymour** [comment]
@big-andy-coates do you think it's worth including some unit tests for the example UDFs? e.g. something similar to the one included [here](https://github.com/confluentinc/ksql/pull/2615/files#diff-6b5533bc02a1354ab6a6855a3794f881) (the UDAF was simplified in that PR some adjustments would need to be made)? Or should we leave that as an exercise for the user?

File: ksql-udf-quickstart/src/main/resources/archetype-resources/src/main/java/ReverseUdf.java (hunk 1 above)
Comment (big-andy-coates) [reply ksql-udf-quickstart/src/main/resources/archetype-resources/src/main/java/ReverseUdf.java]: Yeah, that's true, but only the case if the code is compiled with `-parameters`, which may not be the case.  Better to have the example provide the name IMHO.

**Comment by big-andy-coates** [comment]
@jarekr - hey, yeah I saw your PR - I think your PR also has merit! Adding the example tests to the Archetype might... you never know... encourage users to write unit tests ;)

The only reason I added the 'leave that as an exercise for the reader' comment was to kinda highlight that the provided implementation wasn't really what was needed. Plus I didn't have the time / inclination to actually implement it.  You have - and that's even better!

We're not merging either PR for the 5.2 release. (That ship's sailed). So we can merge both for the 5.2.1 release. I'm easy if you would prefer yours to be merged first.  Either way, between the two PRs I think we've got the archetype into a good state. Thanks for your input!

**Comment by mitch-seymour** [comment]
@big-andy-coates I think you may be referring my PR regarding the unit tests :) but I'm fine with merging these changes for the 5.2.1 release. Let's merge yours first and then I can pull my in changes if that's okay with you

**Pull request merged**